torchtune/data/_collate.py

```diff
@@ -12,7 +12,136 @@
 from torchtune.data._common import CROSS_ENTROPY_IGNORE_IDX
 
 
+def left_pad_sequence(
+    sequences: List[torch.Tensor],
+    batch_first: bool = False,
+    padding_value: float = 0,
+) -> torch.Tensor:
+    """
+    This function is identical to :func:`torch.nn.utils.rnn.pad_sequence`, but
+    instead pads a list of variable length Tensors from the left to the length
+    of the longest sequence.
+
+    Note:
+        This function returns a Tensor of size ``T x B x *`` or ``B x T x *``
+        where `T` is the length of the longest sequence. This function assumes
+        trailing dimensions and type of all the Tensors in sequences are same.
+
+    Args:
+        sequences (List[torch.Tensor]): list of variable length sequences.
+        batch_first (bool): if ``True``, the output will be in ``B x T x *``
+            format, ``T x B x *`` otherwise. Default False.
+        padding_value (float): value for padded elements. Default: 0.
+
+    Returns:
+        Tensor of size ``T x B x *`` if :attr:`batch_first` is ``False``.
+        Tensor of size ``B x T x *`` otherwise
+
+    Example:
+        >>> a = torch.tensor([1, 2, 3])
+        >>> b = torch.tensor([4, 5, 6, 7])
+        >>> c = torch.tensor([8, 9, 10, 11, 12])
+        >>> left_pad_sequence([a, b, c], batch_first=True, padding_value=0)
+        tensor([[ 0,  0,  1,  2,  3],
+                [ 0,  4,  5,  6,  7],
+                [ 8,  9, 10, 11, 12]])
+    """
+    return pad_sequence(
+        map(lambda x: torch.flip(x, dims=[0]), sequences),
+        batch_first=batch_first,
+        padding_value=padding_value,
+    ).flip(dims=[1])
+
+
 def padded_collate(
+    batch: List[Dict[str, List[int]]],
+    *,
+    pad_fn: Callable,
```

Comment:
I'm still on the fence about this, it feels like a case of generalizing too early. In PPO you are only using left pad. I imagine right pad would also be a common case, but why not just specify with a string "left" or "right"?